magvit2_pytorch/magvit2_pytorch.py

```diff
@@ -806,14 +806,16 @@ def __init__(
         super().__init__()
         dim_out = default(dim_out, dim)
         self.maybe_blur = Blur() if antialias else identity
-        self.conv = nn.Conv1d(dim, dim_out, kernel_size, stride = 2, padding = kernel_size // 2)
+        self.time_causal_padding = (kernel_size - 1, 0)
+        self.conv = nn.Conv1d(dim, dim_out, kernel_size, stride = 2)
 
     def forward(self, x):
         x = self.maybe_blur(x, time_only = True)
 
         x = rearrange(x, 'b c t h w -> b h w c t')
         x, ps = pack_one(x, '* c t')
 
+        x = F.pad(x, self.time_casual_padding)
```

Comment:
oops, there was a typo here `time_causal_padding`, uncovered that this morning

Reply:
Oops sorry bout that, cleaned up a bit after testing

Reply:
npnp